extension/src/experiments/model/index.ts

```diff
@@ -324,13 +324,15 @@ export class ExperimentsModel extends ModelWithPersistence {
   }
 
   public getExperimentsWithCheckpoints(): ExperimentWithCheckpoints[] {
-    return this.getExperiments().map(experiment => {
-      const checkpoints = this.getCheckpointsWithType(experiment.id)
-      if (!definedAndNonEmpty(checkpoints)) {
-        return experiment
-      }
-      return { ...experiment, checkpoints }
-    })
+    return this.getExperiments()
+      .filter(({ queued }) => !queued)
+      .map(experiment => {
+        const checkpoints = this.getCheckpointsWithType(experiment.id)
+        if (!definedAndNonEmpty(checkpoints)) {
+          return experiment
+        }
+        return { ...experiment, checkpoints }
+      })
```

Comment:
What about the following, to get rid of the filter and map?
```
  public getExperimentsWithCheckpoints(): ExperimentWithCheckpoints[] {
    const experimentsWithCheckpoints: ExperimentWithCheckpoints[] = []
    const experiments = this.getExperiments()

    for (const experiment of experiments) {
      const { queued, id } = experiment
      const checkpoints = !queued && this.getCheckpointsWithType(id)

      if (definedAndNonEmpty(checkpoints)) {
        experimentsWithCheckpoints.push({
          ...experiment,
          checkpoints
        })
      }
    }

    return experimentsWithCheckpoints
  }
```

And in `array.ts`:
```
export const definedAndNonEmpty = (
  maybeArray: readonly unknown[] | undefined | false
): maybeArray is unknown[] => {
  return !!(maybeArray && maybeArray.length > 0)
}
```

Reply:
I think 
```
      const checkpoints = !queued && this.getCheckpointsWithType(id)

      if (definedAndNonEmpty(checkpoints)) {
```
needs to be 

```      
      if (queued){
        continue
      }
      const checkpoints = this.getCheckpointsWithType(id)

      if (!definedAndNonEmpty(checkpoints)) {
        experimentsWithCheckpoints.push(experiment)
        continue
      }

      ...
```

Reply:
I'll update to simplify 👍🏻.